### DISCUSSION TITLE
RaspberryMatic-3.67.10.20230108-86d4e2-rpi3 NightlyBuild : HmIP-HAP-xxx

### DISCUSSION THREAD

**Discussion by ghost** [OpenCCU/OpenCCU#2150] [🇩🇪 Fragen&Antworten] [answered]
HmIP-HAP JS1 hat den Status "Hide" in der Registerkarte GERÄT? ist das irgendein Fehler?
![Animation2](https://user-images.githubusercontent.com/81189856/211215306-2e1fe2cb-4628-448f-a14f-4b78f901154e.gif)

**Comment by jens-maus** [comment, marked as answer]
Gute Frage. Wie verhält sich denn ein Standard `HmIP-HAP` (ohne `JS1`) hier? Und es gibt ja zusätzlich wohl auch noch den Subtype `HmIP-HAP-B1`. Wie steht es um diesen? Bei einem `HmIPW-DRAP` ist es ja auch so das man den Kanal `:0` nur bekommt wenn man diesen explizit einblenden lässt via dieses Buttons.

**Reply by jp112sdl** [reply]
Ich sag gleich: Ich bau keinen Patch ;-)

Nur als Hint: Schau mal in der `/www/rega/esp/functions.fn`

```
      if ((oDevHssType != "HmIP-CCU3") && (oDevHssType != "HmIP-HAP") && (oDevHssType != "HmIP-HAP-B1") && (oDevHssType != "RPI-RF-MOD") && (oDevHssType != "HmIP-RFUSB")) {                     
        bShowChannel = false;             
      }     
```

Dort fehlt der JS1.
Achtung: `HmIP-HAP JS1` mit Leerzeichen, ohne Bindestrich, also nicht wie beim `-B1` ;-)

**Reply by jens-maus** [reply]
Na gut das du selbst nochmal drüber geschaut hast. Hatte ich auch schon gemacht weil ich ähnliches vermutet hatte aber keinen Treffer in meinen grep aufmachen konnte. Dann bau ich dieses mal wohl nen kurzen Patch dafür! Danke nochmal.

**Reply by jens-maus** [reply]
Ok, war doch kein eQ3 upstream bug, sondern der HmIP-RFUSB Support patch war schuld. Siehe: https://github.com/jens-maus/RaspberryMatic/commit/45516e159ea377d970b77151cf6fdde33a95fec6 Sollte also mit dem nächsten nightly build bzw. der nächsten offiziellen Version dann repariert sein. Danke nochmal @jp112sdl

**Reply by ghost** [reply]
Danke dir auch @jp112sdl @jens-maus

**Comment by ghost** [comment]
HmIP-HAP Kanal: 0 wird korrekt angezeigt. HmIP-HAP-B1,HmIPW-DRAP  habe ich nicht ( vielleicht schreibt ja noch jemand )
PS. Kanal:0 von CCU3 wird ebenfalls standardmäßig angezeigt

![Animation2](https://user-images.githubusercontent.com/81189856/211217029-500afa3e-c18b-4665-9e42-16c5cf2807c7.gif)
